src/main/java/com/uber/jenkins/phabricator/conduit/Differential.java

```diff
@@ -33,44 +32,15 @@
 import java.io.IOException;
 import java.net.MalformedURLException;
 import java.net.URL;
-import java.util.HashMap;
-import java.util.Map;
 
 public class Differential {
-    private final JSONObject rawJSON;
-    private final LauncherFactory launcher;
-    private final String conduitToken;
-    private final String arcPath;
+    private JSONObject rawJSON;
 
-    public Differential(String diffID, LauncherFactory launcher, String conduitToken, String arcPath) throws IOException, InterruptedException, ArcanistUsageException {
-        this.conduitToken = conduitToken;
-        this.arcPath = arcPath;
-        this.launcher = launcher;
-        Map params = new HashMap<String, String>();
-        params.put("ids", new String[]{diffID});
+    private final DifferentialClient client;
 
-        JSONObject query = this.callConduit("differential.querydiffs", params);
-        JSONObject response;
-        try {
-            response = query.getJSONObject("response");
-        } catch (JSONException e) {
-            e.printStackTrace();
-            throw new ArcanistUsageException(
-                    String.format("No 'response' object found in conduit call: (%s) %s",
-                            e.getMessage(),
-                            query.toString(2)));
-        }
-        try {
-            this.rawJSON = response.getJSONObject(diffID);
-        } catch (JSONException e) {
-            e.printStackTrace();
-            throw new ArcanistUsageException(
-                    String.format("Unable to find '%s' key in response: (%s) %s",
-                            diffID,
-                            e.getMessage(),
-                            response.toString(2)));
-
-        }
+    public Differential(DifferentialClient client) throws ArcanistUsageException, IOException, InterruptedException {
```

Comment:
Let's just take in the differential json here. Taking in the client and caching it is not necessary. Let's just make differential a model object that can fetch various pieces of data.